src/execute_order.rs

```diff
@@ -1,6 +1,111 @@
+use crate::ref_finance::{ref_finance, Action, SwapAction, TokenReceiverMessage};
+use crate::utils::{ext_market, ext_token, NO_DEPOSIT};
 use crate::*;
+use near_sdk::env::current_account_id;
+use near_sdk::{ext_contract, is_promise_success, Gas};
+use std::collections::VecDeque;
+
+#[ext_contract(ext_self)]
+trait ContractCallbackInterface {
+    fn remove_liquidity_for_execute_order_callback(&self, order: Order, order_id: U128);
+}
 
 #[near_bindgen]
 impl Contract {
-    pub fn execute_order(&self, order_id: U128) {}
+    pub fn execute_order(&self, order_id: U128) -> PromiseOrValue<U128> {
+        assert!(
```

Comment:
with this check you run `get_order_by_id` twice. You can do `let order = self.get_order_by_id(order_id.0 as u64)`, then `require! (order.is_some ...)` and then unwrap